### ISSUE TITLE
False-positive const-correctness warning would cause compiler error (functor with default args)

### ISSUE THREAD

**Issue by hgrattenthaler** [llvm/llvm-project#107281]
The following code (minimal, marked otherwise) causes clang-tidy to suggest a const-fix which obviously does not compile. It seems to be related to calls to functions taking default arguments of type `std::function<>`. Does **not** reproduce if you take out any of the following:
* the call to `functor`, utilizing the default value
* the argument of `functor` having a default value
* changing the type of the argument of `functor` to any primitive types

```C++
#include <functional>

template <typename function_t = std::function<void()>>
void functor(function_t && f = [](){}) { 

  //not minimal, but isolates the erroneous const-correctness warning
  static_cast<void>(f);
}

int main( ) {
  
  functor();
  bool x = false; //<-- erroneous warning: variable 'x' of type 'bool' can be declared 'const'
  x = true;
  
  //not minimal, but isolates the erroneous const-correctness warning
  static_cast<void>(x);
}
```

**Expected behavior**: clang-tidy does not warn

**Comment by EugeneZelenko** [comment]
Could you please try 19 or `main` branch? https://godbolt.org should be helpful.

**Comment by hgrattenthaler** [comment]
@EugeneZelenko I'll try 19 tomorrow. Note this is a clang-tidy issue, if this is a misunderstanding (see tags).

**Comment by hgrattenthaler** [comment]
I tried it with `Debian LLVM version 19.1.0`, unfortunately, the issue persists.

**Comment by chrchr-github** [comment]
https://godbolt.org/z/9EKb1Mor6
